documentation/docs/developers/connectors/task4.md

```diff
@@ -1,8 +1,178 @@
+
+!!! note
+    The integration of the `AsyncThrottler` into the connector is entirely optional, but it is recommended to enable a better user experience as well as allowing users to manually configure the usable rate limits per Hummingbot instance.
+
+### Types of Rate Limits
+
+There are several types of rate limits that can be handled by the `AsyncThrottler` class. The following sections will detail(with examples) how to initialize the necessary `RateLimit` and how to the throttler is consumed by the connector for each of the different rate limit types. 
+
+!!! warning
+    It is important to identify the exchange's rate limit implementation.
+
+#### 1. Rate Limit per endpoint
+
+This refers to rate limits that are applied on a per endpoint basis. For this rate limit type, the key information to retrieve for each endpoint would be its assigned **limit** and **time interval**.
+
+!!! note
+    Examples of existing connectors that utilizes this rate limit implementation are:</br>
+      (1) Kucoin</br>
+      (2) Crypto.com</br>
+
+##### Configuring Rate Limits
+
+  As mentioned above, the key information to retrieve from the exchange are the `limit` and `time_interval`(in seconds) of each endpoint. We will be referencing the Crypto.com connector as an example for exchanges that implement rate limits per endpoint.
+
+!!! note
+    Rate Limits for Crypto.com can be found [here](https://exchange-docs.crypto.com/spot/index.html#rate-limits).
+
+  All the rate limits are to be initialized in the `crypto_constants.py` file.
+
+  ```python
+  RATE_LIMITS = [
+      RateLimit(limit_id=CHECK_NETWORK_PATH_URL, limit=100, time_interval=1),
+      RateLimit(limit_id=GET_TRADING_RULES_PATH_URL, limit=100, time_interval=1),
+      RateLimit(limit_id=CREATE_ORDER_PATH_URL, limit=15, time_interval=0.1),
+      RateLimit(limit_id=CANCEL_ORDER_PATH_URL, limit=15, time_interval=0.1),
+      RateLimit(limit_id=GET_ACCOUNT_SUMMARY_PATH_URL, limit=3, time_interval=0.1),
+      RateLimit(limit_id=GET_ORDER_DETAIL_PATH_URL, limit=30, time_interval=0.1),
+      RateLimit(limit_id=GET_OPEN_ORDERS_PATH_URL, limit=3, time_interval=0.1),
+  ]
+  ```
+
+!!! note
+    `time_interval` here is in seconds. i.e. The rate limits for `CREATE_ORDER_PATH_URL` is 15 requests every 100ms
+
+#### 2. Rate Limit Pools
+
+Rate limit pools refer to a group of endpoints that consumes from a single rate limit. For this rate limit type, the key information to retrieve for each endpoint are its assigned pool(s) and its respective limit and time interval.
+
+!!! note
+    Examples of existing connectors that utilizes this rate limit implementation are:</br>
+      (1) Binance, Binance Perpetual</br>
+      (2) Ndax</br>
+
+##### Configuring Rate Limits
+
+An example of an exchange implementing a this can be seen in the Ndax connector.
+
+  All the rate limit are initialized in the `ndax_constants.py` file. 
```

Comment:
It's worth mentioning that this is the case in general, but it's possible to have complex rate limits that are created dynamically, and to give bybit as an example